project_template/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)
+
+project(az_exampleshortname
+    LANGUAGES C CXX
+    VERSION 0.1.0)
+
+option(az_exampleshortname_with_docs "Build documentation" ON)
+option(BUILD_TESTING "Build tests" ON)
+
+
+# if needed: add ./cmake to the module path
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 17)
+
+# default to hidden symbols when building shared libraries
+# on unix systems
+set(CMAKE_C_VISIBILITY_PRESET hidden)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
+
+set_property(DIRECTORY PROPERTY COMPILE_OPTIONS
+    $<$<COMPILE_LANG_AND_ID:C,MSVC>:/W4 /wd4201>
+    $<$<COMPILE_LANG_AND_ID:C,Clang>:-Xclang -Wall -Wextra -Wno-microsoft-anon-tag -Wdocumentation>)
+
+add_subdirectory(external)
+
+if(NOT DEFINED ECM_MODULE_PATH)
+    find_package(ECM REQUIRED CONFIG)
+endif()
+list(APPEND CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
+
+
+file(GLOB_RECURSE az_exampleshortname_src CONFIGURE_DEPENDS src/*.c)
+file(GLOB_RECURSE az_exampleshortname_test_src CONFIGURE_DEPENDS test/*.cpp)
+add_library(az_exampleshortname)
+add_library(az::exampleshortname ALIAS az_exampleshortname)
+include(GenerateExportHeader)
+generate_export_header(az_exampleshortname 
+    EXPORT_FILE_NAME inc/az/exampleshortname/export.h)
+target_sources(az_exampleshortname PRIVATE ${az_exampleshortname_src})
+target_include_directories(az_exampleshortname PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/inc>
+    $<INSTALL_INTERFACE:include>
+    PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/inc/az/exampleshortname
+        ${CMAKE_CURRENT_BINARY_DIR}/inc/az/exampleshortname)
+if(BUILD_TESTING)
+    include(CTest)

```

Comment:
What more does this do than `enable_testing()` here? Why do we need that extra functionality?

Reply:
I thought it was better to include it instead of enable_testing, I can change it though.

Reply:
narrator: it was not. Robert is right again.